blogposts/2020/announcing-sourcegraph-3.17.md

```diff
@@ -0,0 +1,257 @@
+
+
+### Automatic precise code intelligence
+
+<div class="container my-4 video-embed embed-responsive embed-responsive-16by9 ">
+    <iframe class="embed-responsive-item" src="https://www.youtube.com/embed/BHYka1CT700?autoplay=0&amp;cc_load_policy=0&amp;start=0&amp;end=0&amp;loop=0&amp;controls=1&amp;modestbranding=0&amp;rel=0" allowfullscreen="" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" frameborder="0"></iframe>
+</div>
+
+An experimental feature on [sourcegraph.com/search](https://sourcegraph.com/search) will automatically create an [LSIF index](https://docs.sourcegraph.com/user/code_intelligence/lsif) for eligible repositories based on their popularity, so that repository will have precise results on hover, definition, and reference operations. We are currently able to index Go repositories containing a go.mod file that does not require additional build steps and are working on expanding the set of eligible repositories to support additional languages and more sophisticated repository structures.
+
+To see this experiment in action, pick a Go repository hosted on GitHub that has yet to be discovered. Possibly something useful with a passionate cult-following. Then, visit that repository in sourcegraph.com/search. After navigating through the code base after a while, you should see the hover tooltips and definition results become more accurate.
+
+Currently, a repository will need 50 navigation events (hovers, jump to definition, find references) to trigger the auto-indexing procedure. We are continually tuning these heuristics to bring precise results to more repositories faster.
+
+
+## Product preview: code insights
+
+TODO: Felix will make a video
+
+The Sourcegraph developer team is prototyping code insights, which will help you answer questions you have about your code, including: 
+
+   * [Code smells](https://martinfowler.com/bliki/CodeSmell.html) over time
+   * Languages used at the organization
+   * Library adoption
+   * Test coverage over time
+   * Tracking a migration
+   * Security anti-patterns
+
+Code insights helps developers be more productive and do their jobs better, and brings the data we have in Sourcegraph together.
+
+We want to understand the bigger questions you have about your code base, and would love to hear from you! Please reach out at [feedback@sourcegraph.com](mailto:feedback@sourcegraph.com) to talk to us.
+
+
+## Sourcegraph developer features
+
+
+### AND/OR query improvements
+
+TODO: Rijnard is making a video
+
+In [Sourcegraph 3.15](https://about.sourcegraph.com/blog/sourcegraph-3.15#experimental-andor-operators-for-search-contents), we introduced AND and OR queries for file content for regexp and structural search modes as an experimental option. Sourcegraph 3.17 now introduces AND and OR queries for all modes (including the newly added literal search mode) and is on by default for all instances (no longer considered experimental).
```

Comment:
@adaaaam re comment above, we should be consistent with `AND/OR` vs `AND` and `OR` here then

Reply:
agreed!

Reply:
Fixed in the copy change I just pushed.